src/raft.rs

```diff
@@ -2082,6 +2092,7 @@ impl<T: Storage> Raft<T> {
         let next_idx = self.raft_log.last_index() + 1;
         prs.restore_snapmeta(meta, next_idx, self.max_inflight);
         prs.get_mut(self.id).unwrap().matched = next_idx - 1;
+        prs.get_mut(self.id).unwrap().committed_index = self.raft_log.committed;
```

Comment:
When handling snapshot, we will come here to set the `progress` for itself first, then restore the raft log, and the `progress` we set here will be stale. Should we move these two lines of code after `self.raft_log.restore(snap);` ?

Reply:
Yes, it should. It seems a bug exists for a long time.

Reply:
I think we can just delete them as this peer must not be leader when handling snapshot, and `progress` for the follower is useless.

Reply:
How about leave it as it is? We can sync the changes with etcd/raft later.

Reply:
🤣 It's no harm because this peer is a follower and follower's progress is useless.